docs/tutorial.rst

```diff
@@ -1,4 +1,209 @@
+       return Application(routes, render_factory=render_factory)
+
+
+   if __name__ == "__main__":
+       app = create_app()
+       app.serve()
+
+
+Let's start from the bottom of this code and work our way up:
+
+- In the last ``if`` clause, we create the application
+  and start it by invoking its ``.serve()`` method.
+
+- Next, we have the ``create_app()`` function
+  where we register the routes of the application.
+  A route associates a URL with a function (*endpoint*)
+  that will process the requests to that URL.
+  In the example, there is only one route where the URL is ``/``
+  and the ``home()`` function is the endpoint.
+
+- The route also sets the template file ``home.html``
+  to render the response.
+  Clastic supports multiple template engines;
+  this example uses `ashes`_.
+  We create a render factory for rendering templates
+  for our chose template engine and tell it where to find
+  the template files.
+
+- The application is created by giving the sequence of routes
+  and the render factory.
+
+- Finally, the ``home()`` function generates the data
+  to be passed to the template.
+  The form will contain two dropdown lists for all available time zones,
+  so we have to pass that list.
+  Here, we construct this as a list of dictionaries
+  where the keys are the location names
+  (the last component of the time zone code),
+  and the full time zone code.
+  The location name will be displayed to the user,
+  whereas the full code will be transmitted as the data.
+  The entries will be sorted by location name.
+
+
+The ``home.html`` template is given below.
+In the selection boxes,
+for each element in the ``zones`` list that is passed as parameter,
+the ``location`` key is used for display
+and the ``zone`` key is used for the value:
+
+.. code-block:: html
+
+   <!DOCTYPE html>
+   <html lang="en">
+   <head>
+     <meta charset="utf-8">
+     <title>Time zone convertor</title>
+   </head>
+   <body>
+     <h1>Time zone convertor</h1>
+     <form action="." method="post">
+       <select name="src">
+         {#zones}
+         <option value="{zone}">{location}</option>
+         {/zones}
+       </select>
+       <input type="datetime-local" name="dt">
+       <select name="dst">
+         {#zones}
+         <option value="{zone}">{location}</option>
+         {/zones}
+       </select>
+       <button type="submit">Show</button>
+     </form>
+   </body>
+   </html>
+
+
+Assuming you've named the Python file ``tzconvert.py``,
+when you run the command ``python tzconvert.py``,
+you can visit the address ``http://127.0.0.1:5000/``
+to see the form.
+
+
+Handling request data
+---------------------
+
+Our form submits the data to the current URL,
+therefore to the same endpoint.
+Now we want to modify the ``home()`` function,
+so that if any such data is submitted,
+the response page will include the result of the conversion:
+
+.. code-block:: python
+
+   def home(request):
+       data = {}
```

Comment:
thinking about it, could `data` be more descriptively named `render_ctx` or `template_ctx`? That's where the data is destined in this case, after all.

Reply:
You're right, and done. I've also arranged the home page function to use that naming scheme.